CMakeLists.txt

```diff
@@ -323,17 +327,27 @@ function(add_iree_mlir_src_dep llvm_monorepo_path)
 
   # Reset CMAKE_BUILD_TYPE to its previous setting.
   set(CMAKE_BUILD_TYPE "${_CMAKE_BUILD_TYPE}" CACHE STRING "Build type (default ${DEFAULT_CMAKE_BUILD_TYPE})" FORCE)
+
+  # Set include directories
+  include_directories(${PROJECT_SOURCE_DIR}/third_party/llvm-project/llvm/include)
+  include_directories(${PROJECT_BINARY_DIR}/third_party/llvm-project/llvm/include)
+  include_directories(${PROJECT_SOURCE_DIR}/third_party/llvm-project/mlir/include)
+  include_directories(${PROJECT_BINARY_DIR}/third_party/llvm-project/llvm/tools/mlir/include)
```

Comment:
yeah no global includes like this - these should not be needed anyway
adding includes like this adds them to every target for every platform for all of the project - they should be added only where needed

Reply:
Well we need to make MLIR-HLO aware of where it finds MLIR. This normally happens via

```CMake
find_package(MLIR REQUIRED CONFIG)
...
include_directories(${LLVM_INCLUDE_DIRS})
include_directories(${MLIR_INCLUDE_DIRS})
```

We cannot use this, as we build a bundled MLIR... or at least I was not able to get it running with a bundled MLIR + passing `MLIR_DIR`. Using an installed MLIR, it's done the way described above. However, we need to pass those includes to some submodules, especially MLIR-HLO. Before this change, the includes were set indirectly via `iree_copts`. Thus, I don't think this change makes it worse, also it is not a big iprovement. Any suggestions who we could do better?

Reply:
`MLIR_INCLUDE_DIRS` and `LLVM_INCLUDE_DIRS` are now used to propagate the include directories to submodules that normally rely on `find_package(MLIR REQUIRED CONFIG)`. Furthermore, the changes to `IREE_COMMON_INCLUDE_DIRS` in `iree_copts.cmake` were reverted and the open TODO can be addressed separately.

Reply:
awesome!